File: examples/tita/src/scenarios/generics.ts
import { echo } from "java:dev.tita.fixtures.Generics";
import { $new as newBox,
$instance$set as boxSet,
$instance$get as boxGet }
from "java:dev.tita.fixtures.Generics$Box";

export function runGenericsScenario() {
  const echoed = echo("tsj");
  const box = newBox();
  boxSet(box, "boxed");
  const boxed = boxGet(box);

  let ok = true;
  if (echoed !== "tsj") {
    ok = false;
  }
  if (boxed !== "boxed") {
    ok = false;
  }

  if (ok) {
    console.log("GENERICS_OK");
  }
  return ok;
}

File: examples/tita/fixtures-src/base/dev/tita/fixtures/Generics.java
package dev.tita.fixtures;

import java.util.List;
import java.util.Map;

public final class Generics {
    private Generics() {
    }

    public static <T extends CharSequence & Comparable<T>> T echo(final T value) {
        return value;
    }

    public static List<? extends Number> nums() {
        return List.of(1, 2, 3);
    }

    public static Map<String, List<Integer>> map() {
        return Map.of("numbers", List.of(1, 2, 3));
    }

    public static final class Box<T> {
        private T value;

        public T get() {
            return value;
        }

        public void set(final T next) {
            this.value = next;
        }
    }
}
